src/Button/Button.component.tsx

```diff
@@ -3,113 +3,602 @@ import * as React from 'react';
+        width: 4,
+        height: 2,
+    },
+    small: {
+        width: 5,
+        height: 2.5,
+    },
+    large: {
+        width: 12.5,
+        height: 3,
+    },
+    xlarge: {
+        width: 12.5,
+        height: 3.5,
+    },
+};
 
-        &:hover {
-            color: ${darken(0.25, colors.savvyCyan.base)};
-        }
-    `,
+const transitionSpeed = `500ms`;
+
+interface ButtonStyles {
+    base: FlattenSimpleInterpolation;
+    disabled: FlattenSimpleInterpolation;
+    hover: FlattenSimpleInterpolation;
+    active?: FlattenSimpleInterpolation;
+    focus?: FlattenSimpleInterpolation;
+    focusOutline?: FlattenSimpleInterpolation;
+}
+
+interface ButtonStylesGroup {
+    [key: string]: ButtonStyles;
+}
+
+const ButtonColorStyles = ({
+    colorTheme,
+    reverse,
+}: {
+    colorTheme: Theme;
+    reverse?: boolean;
+}): ButtonStylesGroup => {
+    const { base, dark, light } = colorTheme;
+
+    if (reverse) {
+        return {
+            primary: {
+                base: css`
+                    border: solid thin ${colors.white.base};
+                    background-color: ${colors.white.base};
+                    color: ${base};
+                `,
+                disabled: css`
+                    border: solid thin ${colors.white.base};
+                    background-color: ${colors.white.base};
+                    color: ${base};
+                    opacity: 0.5;
+                `,
+                hover: css`
+                    border: solid thin
+                        ${transparentize(0.15, colors.white.base)};
+                    background-color: ${transparentize(
+                        0.15,
+                        colors.white.base
+                    )};
+                    color: ${base};
+                `,
+                active: css`
+                    border: solid thin ${colors.white.base};
+                    background-color: ${colors.white.base};
+                `,
+                focus: css`
+                    border: solid thin ${colors.white.base};
+                    background-color: ${colors.white.base};
+                `,
+                focusOutline: css`
+                    box-shadow: 0 0 0 3px ${colors.white.base};
+                `,
+            },
+            outline: {
+                base: css`
+                    border: solid thin ${base};
+                    background-color: transparent;
+                    color: ${base};
+                `,
+                disabled: css`
+                    opacity: 0.5;
+                    border: solid thin ${base};
+                    background-color: transparent;
+                    color: ${base};
+                `,
+                hover: css`
+                    border: solid thin ${base};
+                    background-color: ${base};
+                    color: ${colors.charcoal.light};
+                `,
+                focusOutline: css`
+                    box-shadow: 0 0 0 2px
```

Comment:
px => rem

Reply:
etc. below with the pixels.

Reply:
All of the border sizes in `sizes.theme.ts` are defined in `px`, so I guessed that maybe we didn't want borders/outlines to scale with rems.

Want me to change `sizes.theme` to rems as well?